File: leetcode_198/solution.py
class Solution:
    def rob(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        sa, sb = 0, 0
        for i, x in enumerate(nums):
            if i % 2 == 0:
                sa = max(sa + x, sb)
            else:
                sb = max(sb + x, sa)
        return max(sa, sb)

File: leetcode_198/solution.go
package main

func rob(nums []int) int {
	sa, sb := 0, 0
	for i, x := range nums {
		if i % 2 == 0 {
			if sa + x < sb {
				sa = sb
			} else {
				sa += x
			}
		} else {
			if sb + x < sa {
				sb = sa
			} else {
				sb += x
			}
		}
	}
	if sa > sb {
		return sa
	} else {
		return sb
	}
}